src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -1110,17 +1110,21 @@ class SparseBinaryMatrix {
 
     b << getVersion() << " " << nRows() << " " << nCols() << " ";
 
+    // Rely on operator<< to deal with differing size_type
```

Comment:
I don't understand why the changes in this file are needed. `size_type` is a typedef for `UI1`, a template type that defaults to `UInt32`. So as long as we use NuPIC types as the template types then it should never be an issue.

https://github.com/rcrowder/nupic.core/blob/f280cb442da05f472ef225a2856ec9e03de47d30/src/nupic/math/SparseBinaryMatrix.hpp#L47-L50